frame/vamm/src/lib.rs

```diff
@@ -410,13 +418,21 @@ pub mod pallet {
 			ensure!(!config.base_asset_reserves.is_zero(), Error::<T>::BaseAssetReserveIsZero);
 			ensure!(!config.quote_asset_reserves.is_zero(), Error::<T>::QuoteAssetReserveIsZero);
 			ensure!(!config.peg_multiplier.is_zero(), Error::<T>::PegMultiplierIsZero);
+			let invariant_result =
+				Self::compute_invariant(config.base_asset_reserves, config.quote_asset_reserves);
+			ensure!(
+				invariant_result.is_ok(),
+				Error::<T>::FailedToDeriveInvariantFromBaseAndQuoteAsset
+			);
+			let invariant = invariant_result?;
```

Comment:
Can perhaps use `.map_err(|_| Error::<T>::FailedToDeriveInvariantFromBaseAndQuoteAsset)?` and merge all of this into one assignment?